crates/bevy_hierarchy/src/hierarchy.rs

```diff
@@ -135,15 +135,236 @@ impl<'w> DespawnRecursiveExt for EntityWorldMut<'w> {
     }
 }
 
+/// Disables the given entity and all its children recursively
+#[derive(Debug)]
+pub struct DisableRecursive {
+    /// Target entity
+    pub entity: Entity,
+}
```

Comment:
This functionality raises some questions for me

Should we even have a recursive disable command, or should entity disabling be recursive by default? Probably not, but I don't think either option seems obvious to me.
What would it even mean for an enabled entity to have a disabled parent? If we introduce the possibility for this, how will it affect existing queries? How should `iter_{descendants,ancestors}` behave when only one entity in a family tree is disabled?

Reply:
I agree that disabling the entire hierarchy would be preferred, but I'm not sure how that could be enforced right now without also enforcing the hierarchy, if this we don't want to introduce this ambiguity we should probably wait until that is also enforced.

Reply:
We have the same challenge with despawning. It should probably be recursive by default, but doing so results in a circular dependency between bevy_ecs and bevy_hierarchy. Once bevy_hierarchy is no more, we should re-evaluate.

Reply:
I don't like the idea of introducing this ambiguity. While the default behavior for despawning is potentially surprising, it's somewhat easy to recover from since you're just left with orphaned entities. When disabling a parent, you're leaving the world in a broken state where it's not clear how anything should work.

Reply:
Hmm that's valid. If so, we should make the component type private and move this code into bevy_hierarchy for now.